ddmrp_product_replace/models/stock_buffer.py

```diff
@@ -23,6 +55,42 @@ def _check_demand_product_ids(self):
                     _("Buffered product must be considered as demand.")
                 )
 
+    def _compute_replaced_by_alert_text(self):
+        for rec in self:
+            if rec.replaced_by_id:
+                rec.replaced_by_alert_text = (
+                    _("This product is replaced by %s.")
+                    % rec.replaced_by_id.product_id.display_name
+                )
+            else:
+                rec.replaced_by_alert_text = ""
+
+    def _compute_replacement_for_count(self):
+        for rec in self:
+            rec.replacement_for_count = len(rec.replacement_for_ids)
+
+    @api.depends("replaced_by_id", "replacement_for_ids", "replacement_for_ids.active")
+    def _compute_is_replacement_product(self):
+        for rec in self:
+            rec.is_replacement_product = (
+                rec.replacement_for_ids and not rec.replaced_by_id
+            )
+
+    @api.depends("item_type")
+    def _compute_use_replacement_for_buffer_status(self):
+        for rec in self:
+            rec.use_replacement_for_buffer_status = rec.item_type in [
+                "manufactured",
+                "purchased",
+            ]
+
+    def _compute_procure_recommended_qty(self):
+        res = super()._compute_procure_recommended_qty()
```

Comment:
Also to avoid wasted computing time, I think it would be better to  split the record that need to to be computed by the super method and the others